core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1838,11 +1838,11 @@ private SqlDialect nonOrdinalDialect() {
   }
 
   /** Test case for
-   * <a href="https://issues.apache.org/jira/browse/CALCITE-5044">[CALCITE-5044]
-   * JDBC adapter generates integer literal in ORDER BY, which some dialects
-   * wrongly interpret as a reference to a field</a>. */
```

Comment:
You seem to have stomped on a unit test for CALCITE-5044. Does that mean that you think that the functionality added by CALCITE-5044 is wrong, and should be reverted? 

If so, you should reference CALCITE-5044 in bug CALCITE-5724 and explain why they are in conflict.

(My gut is that they are not in conflict, and you should keep the existing test method and add a new one.)

Reply:
Maybe they can coexist. Add a method `SqlConformance.isSortByLiteralAllowed()`, default true, false in BigQuery. The test would have the old behavior if `isSortByLiteralAllowed` is true, the new behavior if it is false.

Reply:
Applied, let me know if its along the lines of what you had in mind